repose-aggregator/components/filters/valkyrie-authorization/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala

```diff
@@ -96,12 +117,31 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
       case ResponseResult(code, message) =>
         mutableHttpResponse.sendError(code, message)
     }
+
+    //weep for what we must do with this flaming pile, note above where it was wrapped pointlessly just to break the chain
+    mutableHttpResponse.writeHeadersToResponse()
+    mutableHttpResponse.commitBufferToServletOutputStream()
   }
 
   def cacheKey(transformedTenant: String, contactId: String): String = {
     transformedTenant + contactId
   }
 
+  def nonAuthorizedPath(url: => String): Boolean = {
+    val path: String = new URL(url).getPath
+    lazy val onResourceList: Boolean = Option(configuration.getCollectionResources)
+      .getOrElse(new CollectionResources)
+      .getResource.asScala.exists { resource =>
+      resource.getPathRegex.r.findFirstIn(path).isDefined
```

Comment:
I still kind of don't like this. Especially the object instantiation...

How about:
lazy val onResourceList: Boolean = Option(configuration.getCollectionResources)
      .map(_.getResource.asScala)
      .filter(resources => resources.exists(_.getPathRegex.r.pattern.matcher(path).matches()))
      .isDefined

Still not great, but maybe better?